### PR TITLE
Web UI v1beta3 conversion

### PR DESCRIPTION
**Pull request by bcbroussard**
Foundation for v1beta3 api upgrade of web ui
cc/ @lavalamp @bgrant0607 @jackgr

### CODE REVIEW

**Comment by brendandburns** [comment]
Hrm, there are _lots_ of API level chances in switching.  Have we validated that the javascript code can handle it?

**Comment by preillyme** [comment]
@brendanburns I'm going to focus on the API level changes as well. We'll make sure the javascript code is modified as well. I apologize if that wasn't clearly communicated.

**Comment by bgrant0607** [comment]
Yes, I pointed out some of the problems in another PR: https://github.com/GoogleCloudPlatform/kubernetes/pull/8302#discussion_r30381491

**Comment by jackgr** [comment]
We'll also need to update the [configuration collector](https://github.com/google/cluster-insight).

**Comment by lavalamp** [comment]
Yeah, I expect a bunch of js changes before this will work. (note if you tried it like this it probably worked fine because the bin data didn't get regenerated)

**Comment by bcbroussard** [comment]
I'm splitting this PR up into many smaller ones for easier review. 

The code here is ready for review as well.

File: www/app/assets/js/app.js [hunk 1]
```diff
@@ -2195,20 +2244,20 @@ angular.module('kubernetesApp.components.dashboard')
     var pods = {
       "kind": "PodList",
       "creationTimestamp": null,
-      "selfLink": "/api/v1beta1/pods",
+      "selfLink": "/api/v1beta3/pods",
       "resourceVersion": 166552,
-      "apiVersion": "v1beta1",
+      "apiVersion": "v1beta3",
```
Comment (bgrant0607) [inline www/app/assets/js/app.js]: As in the other PR, all of these examples are still using the v1beta1 schema. They need to be totally regenerated using v1beta3.

**Comment by bcbroussard** [comment]
@bgrant0607 I generated new mock files and updated the code. I used on the conversion utility on the earlier commit, which apparently does not account for selfLink. Thanks for the comments.

**Comment by bcbroussard** [comment]
All pages are updated and ready for review. Once these are all in I will do followup PR's removing the v1beta2 api, and also updating to the v1 api endpoint.

**Comment by dchen1107** [comment]
ping one this one since a lot of prs depending on it.

File: www/app/assets/js/app.js [hunk 2]
```diff
@@ -205,7 +206,7 @@ app.controller('TabCtrl', [
       .service('cAdvisorService', ["$http", "$q", "ENV", function($http, $q, ENV) {
         var _baseUrl = function(minionIp) {
           var minionPort = ENV['/']['cAdvisorPort'] || "8081";
-          var proxy = ENV['/']['cAdvisorProxy'] || "/api/v1beta2/proxy/nodes/";
+          var proxy = ENV['/']['cAdvisorProxy'] || "/api/v1beta3/proxy/nodes/";
```
Comment (caesarxuchao) [inline www/app/assets/js/app.js]: Should it be /api/v1beta3/proxy/namespaces/default/nodes?

File: www/app/assets/js/app.js [hunk 3]
```diff
@@ -437,6 +438,54 @@ app.provider('k8sApi',
...
 
+app.provider('k8sv1Beta3Api',
+             function() {
+
+               var urlBase = '';
+               var _namespace = 'default';
+
+               this.setUrlBase = function(value) { urlBase = value; };
+
+               this.setNamespace = function(value) { urlBase = value; };
```
Comment (caesarxuchao) [inline www/app/assets/js/app.js]: looks like it should be {_namespace = value; };

File: www/app/assets/js/app.js (hunk 2 above)
Comment (bcbroussard) [inline www/app/assets/js/app.js]: /nodes isn't namespaced so leaving it as is

**Comment by bcbroussard** [comment]
Thanks @caesarxuchao I've made the setNamespace change, PTAL

File: www/app/assets/js/app.js (hunk 3 above)
Comment (bcbroussard) [inline www/app/assets/js/app.js]: yup, changed

**Comment by caesarxuchao** [comment]
Thanks, @bcbroussard. I'm not familiar with .js, so I just checked the v1beta3 syntax, LGTM.

**Pull request merged**